src/credentials/credential-wallet.ts

```diff
@@ -342,19 +344,40 @@ export class CredentialWallet implements ICredentialWallet {
       type: VerifiableConstants.JSON_SCHEMA_VALIDATOR
     };
 
-    const id =
-      request.revocationOpts.type === CredentialStatusType.SparseMerkleTreeProof
-        ? `${request.revocationOpts.id.replace(/\/$/, '')}/${request.revocationOpts.nonce}`
-        : request.revocationOpts.id;
-
-    cr.credentialStatus = {
-      id,
-      revocationNonce: request.revocationOpts.nonce,
-      type: request.revocationOpts.type
-    };
+    cr.credentialStatus = this.buildCredentialStatus(request);
 
     return cr;
   };
+
+  /**
+   * Builds credential status
+   * @param {CredentialRequest} request
+   * @returns `CredentialStatus`
+   */
+  private buildCredentialStatus(request: CredentialRequest): CredentialStatus {
+    const credentialStatus: CredentialStatus = {
+      id: request.revocationOpts.id,
+      type: request.revocationOpts.type,
+      revocationNonce: request.revocationOpts.nonce
+    };
+
+    switch (request.revocationOpts.type) {
+      case (CredentialStatusType.SparseMerkleTreeProof,
+      CredentialStatusType.Iden3commRevocationStatusV1):
+        credentialStatus.id = `${request.revocationOpts.id.replace(/\/$/, '')}/${
+          request.revocationOpts.nonce
+        }`;
+        break;
+      case CredentialStatusType.Iden3ReverseSparseMerkleTreeProof:
+        credentialStatus.id = request.revocationOpts.issuerState
+          ? `${request.revocationOpts.id}/node?state=${request.revocationOpts.issuerState}`
+          : `${request.revocationOpts.id}`;
+        break;
```

Comment:
For SparseMerkleTreeProof, Iden3commRevocationStatusV1, Iden3ReverseSparseMerkleTreeProof types we have to modify the `id` field. All other types remain unchanged

Reply:
How I understand this regexp cuts end '/'. We should cut the last '/' for the agent endpoint also to be sure that SDK will not create a URL like `https/agent//`

Reply:
@Kolezhniuk @vmidylli about the `default` case. I disagree that we should return an error if we have an unknown type. 
1. What if the user implements a custom revocation type?
2. We have strict typing to protect against dummy users.
3. SDK should be flexible. All validation should be on the user side. Because we cannot know all the use cases of this SDK